frontend/components/site/SitePublishedFilesTable.jsx

```diff
@@ -1,41 +1,141 @@
+
+      filesByPage[currentPage] = files;
+      this.setState({ filesByPage });
+
+      if (!publishedFiles.data.isTruncated) {
+        this.setState({ lastPage: currentPage });
+      }
+    }
+  }
+
+  shouldDisablePreviousPage() {
+    return this.state.currentPage === 0;
+  }
+
+  previousPage() {
+    if (this.state.currentPage > 0) {
+      this.setState({ currentPage: this.state.currentPage - 1 });
+    }
   }
 
-  publishedFiles() {
+  shouldDisableNextPage() {
+    if (this.state.lastPage !== null) {
+      return this.state.currentPage === this.state.lastPage;
+    }
+    return false;
+  }
+
+  nextPage() {
+    const currentPage = this.state.currentPage;
+    const nextPage = currentPage + 1;
+
+    if (this.state.lastPage !== null && currentPage === this.state.lastPage) {
+      // do nothing if already on the known last page
+      return;
+    }
+
+    this.setState({ currentPage: nextPage });
+
+    if (this.state.filesByPage[nextPage]) {
+      // short-circuit if already have next files in state
+      return;
+    }
+
+    // else dispatch action to fetch next page of files
+    const site = { id: this.props.params.id };
     const branch = this.props.params.name;
-    if (this.props.publishedFiles.data && !this.props.publishedFiles.isLoading) {
-      return this.props.publishedFiles.data.filter(file => file.publishedBranch.name === branch);
+    const files = this.state.filesByPage[currentPage];
+    const startAtKey = files[files.length - 1].key;
+
+    publishedFileActions.fetchPublishedFiles(site, branch, startAtKey);
+  }
+
+  shouldShowButtons() {
+    if (this.state.lastPage !== null && this.state.lastPage === 0) {
+      return false;
     }
-    return [];
+    return true;
   }
 
   renderPublishedFilesTable(files) {
-    return (<div>
-      <h3>{this.props.params.name}</h3>
-      <p>Use this page to audit the files that Federalist has publicly published.</p>
-      <table className="usa-table-borderless">
-        <thead>
-          <tr>
-            <th>File</th>
-            <th>Actions</th>
-          </tr>
-        </thead>
-        <tbody>
-          { files.filter(f => !!f.name).map(this.renderBranchFileRow) }
-        </tbody>
-      </table>
-    </div>);
+    const prevButtonClass = `${this.shouldDisablePreviousPage() ? 'usa-button-disabled' : 'usa-button'}`;
+    const nextButtonClass = `pull-right ${this.shouldDisableNextPage() ? 'usa-button-disabled' : 'usa-button'}`;
+    return (
+      <div>
+        <h3>{this.props.params.name}</h3>
+        <p>
+          Use this page to audit the files that Federalist has publicly published.
+          Up to 200 files are shown per page.
+        </p>
+        <table className="usa-table-borderless">
+          <thead>
+            <tr>
+              <th>File</th>
+              <th>Actions</th>
+            </tr>
+          </thead>
+          <tbody>
+            { files.filter(f => !!f.name).map(this.renderBranchFileRow) }
```

Comment:
What causes files to not have names? Just curious 🐈

Reply:
Our redirect object constructs (like for redirecting from `some-site.gov/abc` to `some-site.gov/abc/` end up being "nameless."